File: src/0021_Merge_Two_Sorted_Lists/Merge_Two_Sorted_Lists.py
class Solution(object):
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def mergeTwoLists(self, a1, a2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        if a1==None and a2==None:
            return []
        elif a1==None:
            return a2
        elif a2==None:
            return a1
        s=[]
        index=0
        while a1!=None and a2!=None:
            if a1.val <= a2.val:
                s.append(a1.val)
                a1=a1.next
            else:
                s.append(a2.val)
                a2=a2.next
        
        if a1!=None:
            while a1:
                s.append(a1.val)
                a1=a1.next
        if a2!=None:
            while a2:
                s.append(a2.val)
                a2=a2.next
        return s

File: src/1_100/0021_Merge_Two_Sorted_Lists/Merge_Two_Sorted_Lists.java
﻿/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        ListNode head = new ListNode(-1);
        ListNode p = head;
        while(l1!=null && l2!=null) {
            if(l1.val < l2.val) {
                p.next = l1;
                l1 = l1.next;
            } else {
                p.next = l2;
                l2 = l2.next;
            }
            p = p.next;
        }
        while(l1!=null || l2!=null) {
            p.next = l1!=null? l1 : l2;
            l1 = l1==null? null : l1.next;
            l2 = l2==null? null : l2.next;
            p = p.next;
        }
        return head.next;
    }
}